nats/replica_client.go

```diff
@@ -0,0 +1,402 @@
+package nats
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/benbjohnson/litestream"
+	"github.com/benbjohnson/litestream/internal"
+	natsClient "github.com/nats-io/nats.go"
+	"github.com/nats-io/nats.go/jetstream"
+	"github.com/valyala/bytebufferpool"
+)
+
+const (
+	ReplicaClientType     = "nats"
+	DefaultBucketName     = "litestream"
+	DefaultBucketReplicas = 3
+	DefaultBucketMaxBytes = 0               // unlimited
+	DefaultBucketTTL      = 0 * time.Second // unlimited
+
+	generationsPrefix    = "generations/"
+	generationsPrefixLen = len(generationsPrefix)
```

Comment:
The `generationsPrefixLen` seems extraneous since the [StringHeader](https://pkg.go.dev/reflect#StringHeader) includes the length. I can't imagine there's a performance boost.

Reply:
was using as a constant it a few places,  moved to place of usage.